### ISSUE TITLE
[TODO] Feedback to close M2

### ISSUE THREAD

**Issue by saumyakaran** [buidl-labs/yieldscan-frontend#31]
# TODO

## 🎯 &nbsp; What is the problem that needs to be solved?

Try to be as specific as possible. Think about falsifiable outcomes.

**Refer:** [SMART Goal Examples](https://www.thebalancesmb.com/smart-goal-examples-2951827)

### 🗒 &nbsp; Summary

Final set of TODOs to close M2 (doesn't include css level fixes)

## Global

- [x] For smaller screens, we should add a message saying that we don't support mobile devices yet
- [x] We need to replace all avatar placeholders with identicons
- [x] Truncate the addressed to be something like: "Validator(...GyXaz)", "Nominator(...WWysa)"
- [x] Global controller state - i.e. allow user to make the change using the navbar itself (probably as an option under the "settings" menu)

## Returns calculator

- [x] In returns calculator, advanced validator selection and validator tab stake input - I'm unable to input `0` in the "bond additional funds" input when wallet is connected + I already have some funds bonded
    - And, for the calculator, when I don't input any value in this case, the suggestions don't show up - what if I don't want to change my currently bonded amount?
- [x] It gets difficult to find the already selected validator in the "quick edit" mode
    - [x] We could sort the selected validators to be displayed first by default
- [x] There's no "confirm" button to save my selections
    - [x] Also, the back button shouldn't save my selections, instead discard them - probably after giving me a warning that my selections with be discarded
- [x] Stake amount issue on "quick change" for validators - i.e. the amount per validator based on stake amount not correct.

## Validators tab and advanced selection

- [x] A back button is needed for validators "advanced selection" to return to calculator
    - ~On clicking this - I should be notified that my selections would be lost~ - design feedback, please ignore
- [ ] Input validation is missing in validators tab and advanced validator selection
    - Blocked by design
- [x] In the validator tab, the "return / 100 KSM" doesn't tell me whether the returns are monthly, annual or something else
    - Perhaps this design shall help: https://www.figma.com/file/n65zoIVgwL847PMyerqkLx/YieldScan-v2-Designs?node-id=1326%3A4043
    - [ ] Also, the time period input doesn't have a "confirm" ~and "cancel"~ button (cancel button is not there on the design)
- [x] In the validators tab, if I click on a validator's name to view their profile, the selection state also changes - I wouldn't want this to happen
- [x] In the filters menu (or maybe outside of it), there should be a checkbox which allows me to filter the list to show only my selected validators - since it can be quite difficult to search through the list to view my selections
- [x] The CTA should always be visible on my screen - right now, on smaller screens (like my laptop), it gets hidden due to the height of the rest of the content

## Payment UX

- [x] In the payment UX's final step, there's no button that allows me to confirm my controller account - and probably "reset" should instead be changed to "cancel" which shall revert the state to my previous selection (not my currently used controller on chain or stash in case I'm creating a new nominator)
- [x] Missing back button in payment UX
- [x] Once my payment is complete, I would like to be redirected to the overview tab to see the status of my nominations
    - [x] It is important that these are near realtime, since I would like to see changes due to my actions immediately

## Overview tab

- [x] If I've stopped my nominations, but still have funds bonded, I get the message: "Sorry, no data for your account! :("
    - A better way to handle this case would be to add a CTA to our calculator
- [ ] ~I would like to be informed whether or not my selections are earning optimal returns for me~ [OUT OF SCOPE]
    - If yes, I need some UI level element which informs me that my earnings are optimal (refer: https://www.figma.com/file/n65zoIVgwL847PMyerqkLx/YieldScan-v2-Designs?node-id=1326%3A4524)
    - If not, I would like to have a CTA which allows me to optimize the same (refer: https://www.figma.com/file/n65zoIVgwL847PMyerqkLx/YieldScan-v2-Designs?node-id=1326%3A5025 - also note the flow that it connects to)
- [x] Payment destination state - to be the same as on-chain
    - I would also like to be shown the appropriate destination on the "earnings" card on the overview tab - i.e. "staked" - for compounded, "stash" or "controller"
- [ ] ~I would want all data except earnings history to be realtime~ [OUT OF SCOPE]

## Nominators leaderboard

- [x] For the nominators table, clicking on any nominator should redirect the user to Polkascan's account page for the same (until we're ready with our own profiles)

## Governance

- None

## Validator Profile:
- [x] Validator profile visualization (done by Sahil)

## Council Member Profile:
- [x] Council Member profile visualization (done by Sahil)

## ⏳ &nbsp; ETA

TBD

If you don't specify a timezone, by default IST shall be assumed.

Refer: [ETA culture document](https://docs.google.com/document/d/1ZGa6diesOmx-jSBjm0uj8j4oN75gYqux85qA7DJOm68/edit)

**Comment by akshatbhargava123** [comment]
- [ ] When coming from some other page to calculator, we get the bonded amount - staked Amount in bondExtra input.

**Comment by saumyakaran** [comment]
@akshatbhargava123 Edited the main thread to better depict reality. Is there an ETA for completion of what's left outside of TODOs that are blocked due to design?
